pkg/telemetry/phonehome/gatherer.go

```diff
@@ -83,21 +87,26 @@ func (g *gatherer) identify() {
 		g.telemeter.Track(g.clientType+" Heartbeat", nil, g.opts...)
 	}
 	g.lastData = data
+	g.procs.Done()
 }
 
 func (g *gatherer) loop() {
 	// Send initial data on start:
+	g.procs.Add(1)
 	g.identify()
 	ticker := time.NewTicker(g.period)
+for_loop:
 	for !g.stopSig.IsDone() {
 		select {
 		case <-ticker.C:
+			g.procs.Add(1)
 			go g.identify()
 		case <-g.stopSig.Done():
-			ticker.Stop()
-			return
+			break for_loop
```

Comment:
This could be a simple `return` if you'd use the deferred statement for `ticker.Stop(); g.procs.Done()`